### DISCUSSION TITLE
invalid entity ids in collection

### DISCUSSION THREAD

**Discussion by elk-cloner** [milvus-io/milvus#11766] [Q&A and General discussion]
I'm getting lots of `0` entity ids with this code (Milvus v1.1.1)
```
vector_ids = []
partition_list = collection_info["partitions"]
for partition in partition_list:
    segment_list = partition["segments"]
    for segment in segment_list:
        segment_name = segment["name"]
        status, id_list = milvus.list_id_in_segment(collection_name=collection_name, segment_name=segment_name)
        vector_ids.extend(id_list)
```
but I can't get any vectors with `id=0`
```
vecs = milvus.get_entity_by_id(collection_name=collection_name, ids=[0], partition_tag="_default")
print(vecs)
(Status(code=1, message='Collection or partition is empty'), [])
```
Even when using search api it returns some results with `id=0`(during building the index all the vectors were added with `id > 0`).

**Comment by yhmo** [comment]
Did you assign id for each entity inserted to milvus? 

The "collection_info" is returned by `get_collection_stat()`, there is a time gap between `get_collection_stat()` and `list_id_in_segment()`. If there are some insert/flush action within the time gap, the collection_info may be out of date, so `list_id_in_segment()` may return empty results since the segment may have been deleted or merged.

But if the `list_id_in_segment()` return id=0, that may caused by mistake during insert, maybe the inserted entity was specified to 0.

**Reply by elk-cloner** [reply]
No, there isn't any insert action happening while getting the list_id_in_segment() and beside that I'm 100% sure there wasn't any mistake during inserting data because all the ids are the primary_key of a sql table that is starting from 1. But during inserting data I got some space problem and the disk was filled, is there any chance that this is the reason ?

**Reply by yhmo** [reply]
When you invoke the list_id_in_segment(), is any error log generated under /var/lib/milvus/logs folder?
Milvus 1.x store id array in a separate file, the file name is like "xxxxx.uid". I guess the file is cracked when the disk is filled.
